R/c-wrapper.R

```diff
@@ -64,10 +67,27 @@ make_c_bridge <- function(fsub, strict = TRUE, headers = TRUE) {
     if (uses_rng) "PutRNGstate();",
     ""
   )
-  if (n_protected > 0) {
-    append(c_body) <- glue("UNPROTECT({n_protected});")
+  if (length(return_var_names) == 1L) {
+    if (n_protected > 0) {
+      append(c_body) <- glue("UNPROTECT({n_protected});")
+    }
+    append(c_body) <- glue("return {return_var_names};")
+  } else {
+    return_var_names_un <- unname(return_var_names)
+    append(c_body) <- c(
+      glue("SEXP _ans = PROTECT(Rf_allocVector(VECSXP, {length(return_var_names_un)}));"),
+      imap(return_var_names_un, function(nm, i) {
+        glue("SET_VECTOR_ELT(_ans, {i-1}, {nm});")
+      }),
+      glue("SEXP _names = PROTECT(Rf_allocVector(STRSXP, {length(return_var_names_un)}));"),
+      imap(return_var_names_un, function(nm, i) {
+        glue("SET_STRING_ELT(_names, {i-1}, Rf_mkChar(\"{nm}\"));")
```

Comment:
```suggestion
        glue('SET_STRING_ELT(_names, {i-1}, Rf_mkChar("{nm}"));')
```

Reply:
It seems this approach requires that the list element name is the same as the symbol. i.e., the last expression is like `list(foo = foo, bar = bar)`. 

Is that correct? What do you think about supporting different names for the list elements? E.g., `list(foo = abc, bar = xyz)`.